nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/aws/S3DirectBatchHandler.java

```diff
@@ -0,0 +1,315 @@
+
+        Map<String, Object> batchExtraInfo = batch.getBatchExtraInfo();
+        AssumeRoleResult assumeRoleResult = stsClient.assumeRole(
+                new AssumeRoleRequest().withRoleSessionName(batch.getKey()).withRoleArn(roleArnToAssume));
+        batchExtraInfo
+                .put(Constants.BatchExtraInfo.AWS_SECRET_KEY_ID, assumeRoleResult.getCredentials().getAccessKeyId());
+        batchExtraInfo.put(Constants.BatchExtraInfo.AWS_SECRET_ACCESS_KEY, assumeRoleResult.getCredentials()
+                                                                                           .getSecretAccessKey());
+        batchExtraInfo
+                .put(Constants.BatchExtraInfo.AWS_SESSION_TOKEN, assumeRoleResult.getCredentials().getSessionToken());
+        batchExtraInfo.put(Constants.BatchExtraInfo.BUCKET, bucket);
+        batchExtraInfo.put(Constants.BatchExtraInfo.BASE_KEY, baseBucketKey);
+        batchExtraInfo
+                .put(Constants.BatchExtraInfo.EXPIRATION, assumeRoleResult.getCredentials().getExpiration().toInstant()
+                                                                          .toEpochMilli());
+        batchExtraInfo.put(Constants.BatchExtraInfo.AWS_REGION, awsRegion);
+        batchExtraInfo.put(Constants.BatchExtraInfo.USE_S3_ACCELERATE, s3AccelerationSupported);
+
+        return batch;
+    }
+
+    private boolean hasBatch(String batchId) {
+        return StringUtils.isNotEmpty(batchId) && transientStore.exists(batchId);
+    }
+
+    @Override
+    public Batch newBatch(String batchId) {
+        return initBatch(batchId);
+    }
+
+    @Override
+    protected void init(Map<String, String> configProperties) {
+        if (!containsRequired(configProperties)) {
+            throw new NuxeoException();
+        }
+
+        transientStoreName = configProperties.get(Constants.S3TransientStoreConfig.Properties.TRANSIENT_STORE_NAME);
+
+        String awsSecretKeyId = configProperties.get(Constants.S3TransientStoreConfig.Properties.AWS_SECRET_KEY_ID);
+        String awsSecretAccessKey = configProperties
+                .get(Constants.S3TransientStoreConfig.Properties.AWS_SECRET_ACCESS_KEY);
+        awsRegion = configProperties.get(Constants.S3TransientStoreConfig.Properties.AWS_REGION);
+        roleArnToAssume = configProperties.get(Constants.S3TransientStoreConfig.Properties.AWS_ROLE_ARN);
+        bucket = configProperties.get(Constants.S3TransientStoreConfig.Properties.AWS_BUCKET);
+        baseBucketKey = configProperties
+                .getOrDefault(Constants.S3TransientStoreConfig.Properties.AWS_BUCKET_BASE_KEY, "/");
+
+        s3AccelerationSupported =
+                Boolean.parseBoolean(
+                        configProperties
+                                .getOrDefault(Constants.S3TransientStoreConfig.Properties.USE_S3_ACCELERATION, Boolean.FALSE
+                                        .toString())
```

Comment:
You don't need the default value FALSE here. Every string which are not true (ignoring case) will be false.